modules/editor-one/components/elementor-one-menu-manager.php

```diff
@@ -140,10 +143,108 @@ public function reposition_elementor_menu(): void {
 		ksort( $menu );
 	}
 
+	public function reregister_elementor_menu_for_edit_posts_users(): void {
+		$user = wp_get_current_user();
+		if ( ! $user || ! $user->exists() ) {
+			return;
+		}
+
+		$has_edit_posts = isset( $user->allcaps['edit_posts'] ) && $user->allcaps['edit_posts'];
+		$has_manage_options = isset( $user->allcaps['manage_options'] ) && $user->allcaps['manage_options'];
+
+		if ( ! $has_edit_posts || $has_manage_options ) {
+			return;
+		}
+
+		remove_menu_page( Menu_Config::ELEMENTOR_MENU_SLUG );
+
+		add_menu_page(
+			esc_html__( 'Elementor', 'elementor' ),
+			esc_html__( 'Elementor', 'elementor' ),
+			'edit_posts',
+			Menu_Config::ELEMENTOR_MENU_SLUG,
+			[ Plugin::instance()->settings, 'display_home_screen' ],
+			'',
+			58.5
+		);
+	}
+
+
+	public function set_menu_building_flag(): void {
+		self::$is_building_menu = true;
+	}
+
+	public function clear_menu_building_flag(): void {
+		self::$is_building_menu = false;
+	}
+
+
+	public function remove_all_submenus_for_edit_posts_users(): void {
+		$user = wp_get_current_user();
+		if ( ! $user || ! $user->exists() ) {
+			return;
+		}
+
+		$has_edit_posts = isset( $user->allcaps['edit_posts'] ) && $user->allcaps['edit_posts'];
+		$has_manage_options = isset( $user->allcaps['manage_options'] ) && $user->allcaps['manage_options'];
+
+		if ( $has_manage_options || ! $has_edit_posts ) {
+			return;
+		}
+
+		global $submenu;
+
+		if ( empty( $submenu[ Menu_Config::ELEMENTOR_MENU_SLUG ] ) ) {
+			return;
+		}
+
+		$submenu_items = $submenu[ Menu_Config::ELEMENTOR_MENU_SLUG ];
+
+		foreach ( $submenu_items as $index => $submenu_item ) {
+			if ( 0 === $index ) {
+				continue;
+			}
+
+			$submenu_slug = $submenu_item[2] ?? '';
+			if ( ! empty( $submenu_slug ) ) {
+				remove_submenu_page( Menu_Config::ELEMENTOR_MENU_SLUG, $submenu_slug );
+			}
+		}
+	}
+
 	public function render_editor_page(): void {
 		Plugin::instance()->settings->display_home_screen();
 	}
 
+	public function override_elementor_page_for_edit_posts_users(): void {
+		$user = wp_get_current_user();
+		if ( ! $user || ! $user->exists() ) {
+			return;
+		}
+
+		$has_edit_posts = isset( $user->allcaps['edit_posts'] ) && $user->allcaps['edit_posts'];
+		$has_manage_options = isset( $user->allcaps['manage_options'] ) && $user->allcaps['manage_options'];
+
+		if ( $has_manage_options || ! $has_edit_posts ) {
+			return;
+		}
+
+		$page = filter_input( INPUT_GET, 'page', FILTER_SANITIZE_FULL_SPECIAL_CHARS ) ?? '';
+		if ( Menu_Config::ELEMENTOR_MENU_SLUG !== $page ) {
+			return;
+		}
+
+		$templates_url = admin_url( 'edit.php?post_type=elementor_library&tabs_group=library' );
+		wp_safe_redirect( $templates_url );
```

Comment:
We don't have a home page for editor and contributor users, therefore I am forwarding the page directly to the Saved Templates page.

Wdyt?